sdk/search/azure-search/src/main/java/com/azure/search/SearchServiceAsyncClient.java

```diff
@@ -1300,265 +1009,187 @@ public String getEndpoint() {
-        return restClient
-            .skillsets()
-            .createOrUpdateWithRestResponseAsync(skillset.getName(),
-                skillset, requestOptions, accessCondition, context)
-            .map(Function.identity());
-    }
-
     /**
      * Deletes a cognitive skillset in an Azure Cognitive Search service.
      *
      * @param skillsetName the name of the skillset to delete
      * @return a response signalling completion.
      */
     public Mono<Void> deleteSkillset(String skillsetName) {
-        try {
-            return this.deleteSkillsetWithResponse(skillsetName, null, null)
-                .flatMap(FluxUtil::toMono);
-        } catch (RuntimeException ex) {
-            return monoError(logger, ex);
-        }
+        return deleteSkillsetWithResponse(skillsetName, null, null).flatMap(FluxUtil::toMono);
     }
 
     /**
      * Deletes a cognitive skillset in an Azure Cognitive Search service.
      *
      * @param skillsetName the name of the skillset to delete
      * @param accessCondition the condition where the operation will be performed if the ETag on the server matches or
-     *                        doesn't match specified values
-     * @param requestOptions additional parameters for the operation.
-     *                       Contains the tracking ID sent with the request to help with debugging
+     * doesn't match specified values
+     * @param requestOptions additional parameters for the operation. Contains the tracking ID sent with the request to
+     * help with debugging
      * @return a response signalling completion.
      */
-    public Mono<Response<Void>> deleteSkillsetWithResponse(String skillsetName,
-                                                           AccessCondition accessCondition,
-                                                           RequestOptions requestOptions) {
-        try {
-            return withContext(context -> this.deleteSkillsetWithResponse(skillsetName, accessCondition, requestOptions,
-                context));
-        } catch (RuntimeException ex) {
-            return monoError(logger, ex);
-        }
-    }
-
-    Mono<Response<Void>> deleteSkillsetWithResponse(String skillsetName,
-                                                    AccessCondition accessCondition,
-                                                    RequestOptions requestOptions,
-                                                    Context context) {
-        return restClient
-            .skillsets()
-            .deleteWithRestResponseAsync(skillsetName, requestOptions, accessCondition, context)
-            .map(Function.identity());
-    }
-
-    /**
-     * Determines whether or not the given skillset exists.
-     *
-     * @param skillsetName the name of the skillset
-     * @return true if the skillset exists; false otherwise.
-     */
-    public Mono<Boolean> skillsetExists(String skillsetName) {
```

Comment:
Why do we remove this API?

Reply:
Per request from Bruce, it will be easier to add this back in the future than removing it from our APIs.